src/index.coffee

```diff
@@ -18,26 +18,32 @@ for modName, mod of APP
       AngularServiceGenerator.makeEventConstants ajmModule, eventMappings[modName]
 
 #Wrap the jitsi APP object in angular
-ajmModule.factory 'jitsiApp', (EventAdapter) ->
+ajmModule.provider 'jitsiApp', ->
+  vm = this
   eventsWired = false
-
-  APP.wireUpEvents = ->
-    if !eventsWired
-      console.log "---- Wiring UP Events"
-      #AngularServiceGenerator.wireUpEvents EventAdapter, ajmModule, APP, eventMappings
+  vm.config =
+    wireEvents: true
+
+  vm.wireEvents = (shouldWireEvents) ->
+    vm.config.wireEvents = shouldWireEvents
+
+  vm.$get = (EventAdapter) ->
+    APP.wireUpEvents = ->
+      if !eventsWired && vm.config.wireEvents
```

Comment:
What about moving the check of vm.config.wireEvents to line 44 where wireUpEvents is invoked?  That way wireUpEvents is more like autoWireUpEvents and you can choose if you want to wire up the events later.